libraries/c_sdk/standard/mqtt_v2/src/mqtt_lightweight.c

```diff
@@ -1998,7 +1998,7 @@ MQTTStatus_t MQTT_DeserializeAck( const MQTTPacketInfo_t * pIncomingPacket,
         LogError( ( "pSessionPresent cannot be NULL for CONNACK packet." ) );
         status = MQTTBadParameter;
     }
-    else if( pIncomingPacket->pRemainingData == NULL )
+    else if( ( pIncomingPacket->pRemainingData == NULL ) && ( pIncomingPacket->type != MQTT_PACKET_TYPE_PINGRESP ) )
```

Comment:
Is this from syncing the new library?

Reply:
I will raise a different PR for it.This I fixed earlier before syncing with the new library when I found this bug.

Reply:
No change should be directly made to the library, Ideally, it should be fixed in C-SDK which is the Source of truth and then synced. I think I understand the reason why you had to do it. Without this change it will fail some cases. I think when syncing the library code in the PR that you mentioned, It will be handled.